docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,239 @@ sidebar_position: 7
+:::
+
+### 1. Fork the repository on GitHub
+
+Forking creates an exact copy of the entire codebase to your own GitHub account. This means all it's files and the history of changes. You own this copy (in terms of permissions) and it's completely separate from the original. No matter what changes you make, the original codebase (often referred to as `upstream`) will remain unchanged.
+
+You can fork a repository on GitHub from its web interface by going to the repository's URL. For example, the repository for this documentation is `https://www.github.com/BobaBoard/boba-docs`, where `BobaBoard` is the name of the owner of the repository, and `boba-docs` is the repository's name.
+
+Once you've created a fork, this copy will live somewhere that looks like so: `https://www.github.com/<YOUR_GITHUB_USERNAME>/<REPOSITORY_NAME>`
 
-:::warning
-If you want to use a different GitHub account for BobaBoard, your `origin` URL
-will be different. Follow the instructions in the ["use a different GitHub account" section](#extra-use-a-different-github-account).
+:::info
+The name of your fork doesn't have to match the name of the original project. For this guide, however, we'll assume you have kept the same name.
 :::
 
-At this point, you can proceed with the regular flow for a pull request.
+### 2. Clone your fork to your machine
 
-### 1. Create a branch off of `master` for your new feature
+This means copying the code that is stored in your GitHub account to your PC. It doesn't really matter where you want to do this (make sure you have permissions?), though you might want to create a folder to keep all your BobaBoard-related coding in. For example: `/home/NAME/code-projects/bobaboard`.
 
-In the folder for your **forked** repository (see step 0) run
+:::danger
+
+Windows users using WSL will want to pay extra attention.
+
+> //TODO: be less mysterious about it...
+
+:::
+
+In order to clone the repository you need to tell git where the code you want to copy is stored. Simply put, you need a URL. Your repository's URL is simply the web address you use to view it on GitHub with a `.git` at the end. Meaning that if the URL you can copy from your browser's navigation bar is: `https://github.com/<YOUR_GITHUB_USERNAME>/boba-components` then the URL you'll need to use when cloning is:  `https://github.com/<YOUR_GITHUB_USERNAME>/boba-components.git`.
+
+GitHub actually has a handy button in its interface to get this URL.
+
+> //TODO: add screenshot?
+
+:::caution On HTTPS vs SSH
+There's HTTPS and SSH..?
```

Comment:
Oh God, IDK exactly what to say here. It depends on how the user set up GitHub authentication on their machine. I _think_ GitHub now forces SSH?

You can check which one you need by going to a repo that is setup correctly and doing `git remote -v`, and if the addresses start with `git@github.com:` then that's SSH.

IDK if any of this is useful. Check what you're doing too and let me know :)

Reply:
I use SSH because if you use HTTPS you need to provide a password every time? I can't recall exactly, but it was annoying. I set up mine using the instructions from [The Odin Project](https://www.theodinproject.com/lessons/foundations-setting-up-git) but [GitHub](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys?platform=linux) has similar, but more complicated maybe, documentation we can link.

This was kind of a note to self, to remember to adress it 'cos it seems important, but I don't know

Reply:
I always use https and have never had to provide a password? So now I'm confused.